src/inmanta/deploy/work.py

```diff
@@ -394,7 +396,10 @@ def extend_blocked_on(resource: ResourceIdStr, new_blockers: set[ResourceIdStr])
                 task_priority: Optional[TaskPriority] = self.agent_queues.discard(task)
                 queued.remove(resource)
                 self._waiting[resource] = BlockedDeploy(
-                    task=PrioritizedTask(task=task, priority=task_priority if task_priority is not None else priority),
+                    task=PrioritizedTask(
+                        task=task,
+                        priority=task_priority if task_priority is not None and task_priority <= priority else priority,
```

Comment:
100% optional suggestion that I find reads slightly easier. Feel free to leave as is if you prefer it like that.
```suggestion
                        priority=max(priority, task_priority) if task_priority is not None else priority,
```

Reply:
I believe here is `min` right? The lowest priority is the most urgent

Reply:
But I agree that is much more readable